pallets/rc-migrator/src/lib.rs

```diff
@@ -364,10 +369,64 @@ pub mod pallet {
 
 	impl<T: Config> Pallet<T> {
 		/// Execute a stage transition and log it.
-		fn transition(new_stage: MigrationStage<T::AccountId>) {
+		fn transition(new: MigrationStage<T::AccountId>) {
 			let old = RcMigrationStage::<T>::get();
-			RcMigrationStage::<T>::put(&new_stage);
-			log::info!(target: LOG_TARGET, "[Block {:?}] Stage transition: {:?} -> {:?}", frame_system::Pallet::<T>::block_number(), old, new_stage);
+			RcMigrationStage::<T>::put(&new);
+			log::info!(target: LOG_TARGET, "[Block {:?}] Stage transition: {:?} -> {:?}", frame_system::Pallet::<T>::block_number(), &old, &new);
+			Self::deposit_event(Event::StageTransition { old, new });
+		}
+
+		/// Split up the items into chunks of `MAX_MSG_SIZE` and send them as separate XCM
+		/// transacts.
+		///
+		/// This is done to avoid exceeding the XCM message size limit.
```

Comment:
```suggestion
```suggestion
		/// This is done to avoid exceeding the XCM message size limit.
		///
		/// Note: The order of `items` is not preserved.
		/// The `items` are batched and processed in reverse order for optimization.
```
```

Reply:
reverted